experimental/dds/tree2/src/feature-libraries/flex-tree/lazyNode.ts

```diff
@@ -287,43 +352,23 @@ export abstract class LazyTreeNode<TSchema extends TreeNodeSchema = TreeNodeSche
 						// Ugly casting workaround because I can't figure out how to make TS understand that in this case block
 						// the listener argument only needs to be an AnchorNode. Should go away if/when we make the listener signature
 						// for changing and subtreeChanging match the one for beforeChange and afterChange.
-						listener(anchorNode as unknown as AnchorNode & TreeEvent),
+						listener(anchorNode as unknown as AnchorNode & ITreeEvent),
 				);
 				return unsubscribeFromSubtreeChange;
 			}
-			case "beforeChange": {
-				const unsubscribeFromChildrenBeforeChange = this.#anchorNode.on(
-					"beforeChange",
-					(anchorNode: AnchorNode) => {
-						const treeNode = anchorNode.slots.get(lazyTreeSlot);
-						assert(
-							treeNode !== undefined,
-							0x7d3 /* tree node not found in anchor node slots */,
-						);
-						// Ugly casting workaround because I can't figure out how to make TS understand that in this case block
-						// the listener argument only needs to be a TreeEvent. Should go away if/when we make the listener signature
-						// for changing and subtreeChanging match the one for beforeChange and afterChange.
-						listener({ target: treeNode } as unknown as AnchorNode & TreeEvent);
-					},
-				);
-				return unsubscribeFromChildrenBeforeChange;
-			}
+			case "beforeChange":
 			case "afterChange": {
-				const unsubscribeFromChildrenAfterChange = this.#anchorNode.on(
-					"afterChange",
-					(anchorNode: AnchorNode) => {
-						const treeNode = anchorNode.slots.get(lazyTreeSlot);
-						assert(
-							treeNode !== undefined,
-							0x7d4 /* tree node not found in anchor node slots */,
-						);
-						// Ugly casting workaround because I can't figure out how to make TS understand that in this case block
-						// the listener argument only needs to be a TreeEvent. Should go away if/when we make the listener signature
-						// for changing and subtreeChanging match the one for beforeChange and afterChange.
-						listener({ target: treeNode } as unknown as AnchorNode & TreeEvent);
-					},
+				// Type narrowing and casts to make TS happy, because the switch() statement having entered this branch doesn't
+				// seem to be enough.
+				const localEventName: "beforeChange" | "afterChange" = eventName;
+				this.#listeners[localEventName].push(
+					listener as EditableTreeEvents["beforeChange"],
```

Comment:
It's shorter but I'm not totally sure I prefer it to the version with a Map in https://github.com/microsoft/FluidFramework/pull/18538/files/44a0991e53f88e714c80a357a9797120715e0333, mainly because of the type narrowing and cast.